openra/utility.py

```diff
@@ -257,45 +253,39 @@ def map_upgrade(mapObject, engine, parser=list(reversed(list(settings.OPENRA_VER
                 LintCheck(transac, "", parser)
 
                 print("\nRunning SHPtoGIF generator, in case there are SHP files\n")
-                #shp = multiprocessing.Process(target=GenerateSHPpreview, args=(item, parser,), name='SHPtoGIF')
-                #shp.start()
 
             if os.path.isdir(ora_temp_dir_name):
                 shutil.rmtree(ora_temp_dir_name)
 
-    os.chdir(currentDirectory)
     return upgraded_maps
 
 
 def recalculate_hash(item, fullpath="", parser=settings.OPENRA_ROOT_PATH + list(reversed(list(settings.OPENRA_VERSIONS.values())))[0]):
 
-    currentDirectory = os.getcwd() + os.sep
-    os.chdir(parser + "/")
+    currentDirectory = settings.BASE_DIR
 
     if fullpath == "":
 
-        path = currentDirectory + 'openra/data/maps/' + str(item.id) + '/'
+        path = os.path.join(currentDirectory, 'openra', 'data', 'maps', str(item.id))
         filename = ""
         Dir = os.listdir(path)
         for fn in Dir:
             if fn.endswith('.oramap'):
                 filename = fn
                 break
         if filename == "":
-            os.chdir(currentDirectory)
             print('Failed to recalculate hash for %s: %s' % (item.id, 'can not find map'))
             return {'response': 'can not find map', 'error': True, 'maphash': 'none'}
-        fullpath = path + filename
+        fullpath = os.path.join(path, filename)
 
     os.chmod(fullpath, 0o444)
```

Comment:
I don't think `--map-hash` should ever modify a map (it may have used to in the past), so it may be safe to remove the this and its counterpart below.  My assumption would need to be tested first though!

Reply:
These chmods were the cause of the recent upload 500 errors - the script must have died before resetting the write bit, which then stopped all subsequent uploads from working.

Reply:
It would be good if the tmpfile had some uniqe name in it instead of the same file, Because if something stops, it won't block future uploads at least

Reply:
Well it's only changing the permissions on the map file itself. I'd presume that this is to prevent the file getting overwritten or updated while it's working on it.

Writing to a unique name in a temporary directory seems like a good way forward.

Could this be looked into as a separate issue, as it isn't a new thing for this PR, and should really be given specific attention?